### PR TITLE
add possibility to set a due date for credit transfert

### PR DESCRIPTION
**Pull request by senseicoder**
My customer want to set a due date for credit, the SEPA file seems to be valid. Be kind, only my second pull request.

### CODE REVIEW

**Comment by monofone** [comment]
Thx for your contribution, you don't need to close a pull request when tests fail, just push a new commit to the same branch the PR is based on and github will pick this up and run all tests again. Take a look at your closed PR #67 you will see that it already shows your new commit. A better habit would be to create a separate branch for a PR instead of the master. But you will learn all this when you propose more changes to other projects 👍 .

File: lib/Digitick/Sepa/TransferFile/Facade/BaseCustomerTransferFileFacade.php [hunk 1]
```diff
@@ -64,4 +64,20 @@ public function asXML()
 
         return $this->domBuilder->asXml();
     }
+
+    /**
+     * @return DateTime  
+    */
+    public function HandleDueDate(array $paymentInformation)
```
Comment (monofone) [inline lib/Digitick/Sepa/TransferFile/Facade/BaseCustomerTransferFileFacade.php]: Methods start lowerCase

Comment (senseicoder) [reply lib/Digitick/Sepa/TransferFile/Facade/BaseCustomerTransferFileFacade.php]: Sorry, I should have read more code and check for style. Thanks for the welcome :-D

Comment (senseicoder) [reply lib/Digitick/Sepa/TransferFile/Facade/BaseCustomerTransferFileFacade.php]: And the advices. I was more in the subversion team, and I do not understand the whole Github system yet.

File: lib/Digitick/Sepa/TransferFile/Facade/CustomerDirectDebitFacade.php [hunk 2]
```diff
@@ -61,15 +61,7 @@ public function addPaymentInfo($paymentName, array $paymentInformation)
...
-        if (isset($paymentInformation['dueDate'])) {
-            if ($paymentInformation['dueDate'] instanceof \DateTime) {
-                $payment->setDueDate($paymentInformation['dueDate']);
-            } else {
-                $payment->setDueDate(new \DateTime($paymentInformation['dueDate']));
-            }
-        } else {
-            $payment->setDueDate(new \DateTime(date('Y-m-d', strtotime('now + 5 days'))));
-        }
+        $payment->setDueDate($this->HandleDueDate($paymentInformation));
```
Comment (monofone) [inline lib/Digitick/Sepa/TransferFile/Facade/CustomerDirectDebitFacade.php]: ähm .. you need to update the calls of the renamed method as well ...

File: lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php [hunk 3]
```diff
@@ -34,6 +34,8 @@ public function addPaymentInfo($paymentName, array $paymentInformation)
             $paymentInformation['debtorName']
         );
 
+        $payment->setDueDate($this->HandleDueDate($paymentInformation));
```
Comment (monofone) [inline lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php]: ähm .. you need to update the calls of the renamed method as well ...
Thinking longer about the name it should read createDueDateFromPaymentInformation this implies a creation of the date if it is not set on the payment information.

**Comment by aquariuz** [comment]
I'd also like the option of setting a due date for credit transfers. If I understand it correctly, @senseicoder has already done most of the work, except for changing the method name into `createDueDateFromPaymentInformation`. I'd like to help but how to proceed?

**Comment by monofone** [comment]
@aquariuz all what have to be done ist the consistent renaming of the Method. As far as I remember the calls to the method miss the renaming, thats all.

**Comment by monofone** [comment]
I'll close this one in favour of #76, thx for your help

**Comment by senseicoder** [comment]
Sorry, I forgot to do the last step. Thanks Monofone, and thanks Aquariuz.

**Pull request closed without merging**